data/transactions/logic/eval.go

```diff
@@ -159,6 +162,9 @@ type LedgerForLogic interface {
 	DelGlobal(key string) error
 
 	GetDelta(txn *transactions.Transaction) (evalDelta basics.EvalDelta, err error)
+
+	AppendLog(txn *transactions.Transaction, value string) error
```

Comment:
This is because I need to call a txn method to get the app index. https://github.com/shiqizng/go-algorand/blob/opcode/log/ledger/applications.go#L243